R/cov_mx.R

```diff
@@ -1,12 +1,66 @@
+#' Covariance matrix
+#'
 #' `cov_mx` function calculates the covariance matrix of the two
 #'  variables and automatically deals with the missing value
 #'
+#' @usage cov_mx(x)
+#'
 #' @param x a matrix that contains one or more than one random variable
 #'
-#' @return  matrix that contains the covariance between random variables in the input matrix
+#' @return z matrix that contains the covariance between random
+#' variables in the input matrix
+#'
 #' @export
 #'
-#' @examples
+#' @examples cov_mx(matrix(c(1, 2, NA, 4, 5, 6, 7, 8), 4))
 cov_mx <- function(x){
-  message("running cov_mx")
+
+  # check whether input vectors are valid or not
+  if(is.null(x)){
+    stop("Input cannot be empty.")
+  }
+
+  # the input data should be numeric
+  if(!(is.numeric(x) || is.logical(x))) stop("'x' must be numeric.")
+  stopifnot(is.atomic(x))
+
+   # return ERROR if the inputs are empty
+  if(length(x) == 0L){
+    stop("'x' must be non-empty.")
+  }
+
+  # return NA if The length of input equal to 1
+  else if (length(x) == 1L){
+    return (NA)
+  }
+
+  # return error if matrix is not 2D
+  if (dim(x)[2] == 1){
+    stop("'x' must be 2D matrix")
```

Comment:
If input dimension is not 2, then the function should `return (NA)`.

Reply:
Also, change the corresponding test cases to `expect_true(is.na(cov_mx(single_row_x)))`